### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by erikgarciam**
- Modify DataServlet.java to be able to handle comments.
- Add Form to homepage to post comment
- Add break to script.js logic to display comments

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -24,25 +24,68 @@
...
-    
+*/
+  
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         
-        String json = new Gson().toJson(mylist);
-
+    
+        //String json = convertToJson(commentlist);
```
Comment (avenkat-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: Can you remove this comment.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -24,25 +24,68 @@
...
+        
+    }
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String text = getParameter(request, "text-input", "");
+
+        // Break the text into individual words.
+        // Name, LastName,Comment
+        String[] words = text.split("\\s*,\\s*");
+        templist.clear();
```
Comment (avenkat-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:62]: Rename to something that gives more clarity on the variable.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -24,25 +24,68 @@
...
+        json += "\"" + wordspass.get(0) + "\"";
+        json += ", ";
+        json += "\"Last Name\": ";
+        json += "\"" + wordspass.get(1) + "\"";
+        json += ", ";
+        json += "\"Comment\": ";
+        json += wordspass.get(2);
+        json += "}";
+        return json;
+    }
```
Comment (avenkat-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:81]: This isn't used, so should be deleted?

Comment (erikgarciam) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:81]: Yeah I deleted it but the cloud shell didn't save I guess..

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (erikgarciam) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:62]: will do.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -24,25 +24,44 @@
...
-        
-
-    public DataServlet(){
-        mylist.add("Hello this");
-        mylist.add("is a");
-        mylist.add("test");
-    }
-    
+    ArrayList<String> commentlist = new ArrayList<String>();
+    ArrayList<String> comment = new ArrayList<String>();
```
Comment (avenkat-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:31]: Why do you need both commentList and comment? It seems to me like having a Comment class and appending to the commentList would accomplish the same thing, without having to maintain the comment state across the get and post.

Comment (erikgarciam) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:31]: I didn't know you could use classes in gson conversion.

Comment (avenkat-google) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:31]: You should be able to do something like this: https://github.com/google/gson/blob/master/UserGuide.md#object-examples

**Comment by avenkat-google** [comment]
Also can we have some way to validate the input so that the server doesn't crash? (ex. input without commas would crash the servlet). I also recommend splitting up the input boxes for each requested input.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (erikgarciam) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Finished

**Comment by erikgarciam** [comment]
I added a check empty boolean class to determine if the parameters are empty so it skips adding it to the list of comments

**Comment by erikgarciam** [comment]
I just also updated index.html to make it require to fill out each field for extra security.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -24,25 +24,67 @@
...
+    public void setLName(String lastname_){lastname = lastname_;}
+    public void setComment(String comment_){comment = comment_;}
+}
+
+class CheckEmpty{
+    private boolean empty = false;
+
+    public void setEmpty(boolean empty_){empty = empty_;}
+    public boolean isEmpty(){return empty;}
+}
```
Comment (avenkat-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: I'm not quite sure why you need this? Shouldn't the get be fine even if the user doesn't input a comment?

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -34,10 +34,22 @@ <h1 style="text-align:center;">Erik's Portfolio</h1>
...
       <h2><a href="index3.html">My Coursework and Projects.</a><h2>
       <h2><a href="index4.html">Interests and Hobbies.</a><h2>
+      
+      <h1>Comment</h1>
+
+    <form action="/comments" method="POST" id="form1" >
 
-      <button onclick="getMessage()">Fetch some Data!</button>
-      <div id="data-container"></div>
+      <p>Enter a comma-separated list of words:</p>
```
[the remark this answers is not in the document]
Comment (erikgarciam) [reply portfolio/src/main/webapp/index.html]: I updated this in my recent commit of index html.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (erikgarciam) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: Yeah, I think i'm just going to remove CheckEmpty from the code

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -24,25 +24,67 @@
...
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         
-        String json = new Gson().toJson(mylist);
-
+        Gson gson =new Gson();
+        String json = gson.toJson(commentcontainer);
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        if(json.length()> 38 || checkempty.isEmpty() == false){
```
[the remark this answers is not in the document]
Comment (erikgarciam) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I just removed the CheckEmpty class.

**Pull request merged**